gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -410,6 +436,76 @@ ChannelCredentials createMtlsChannelCredentials() throws IOException, GeneralSec
     return null;
   }
 
+  @VisibleForTesting
+  boolean isGoogleS2AEnabled() {
+    String S2AEnv = envProvider.getenv(S2A_ENV_ENABLE_USE_S2A);
+    boolean isS2AEnv = Boolean.parseBoolean(S2AEnv);
+    if (isS2AEnv) {
+      return true;
+    }
+    return false;
+  }
+
+  @VisibleForTesting
+  boolean shouldUseS2A() {
+    // If EXPERIMENTAL_GOOGLE_API_USE_S2A is not set to true, skip S2A.
+    if (!isGoogleS2AEnabled()) {
+      return false;
+    }
+    // If {@link mtlsEndpoint} is not set, skip S2A. S2A is also skipped when there is endpoint
+    // override. Endpoint override is respected when the {@link endpoint} is resolved via AIP#4114,
+    // see EndpointContext.java
+    if (endpoint != mtlsEndpoint) {
+      return false;
+    }
```

Comment:
The sole reason I plumbed the mtlsEndpoint was to perform this check. I think there could possibly be a better way which would allow us to avoid making the change to the TransportChannelProvider interface and all the implementors. 

The reason for performing this check is to only use S2A when we are connecting to a mTLS endpoint. The parallel check in Go: https://github.com/googleapis/google-cloud-go/blob/main/auth/internal/transport/s2a.go#L117

Reply:
Technically, we could just use endpoint, since if endpoint is resolved to use mtls, it will be the mtls endpoint. So down below, where I have:
```
 // Create the channel using S2A-secured channel credentials.
builder = Grpc.newChannelBuilder(mtlsEndpoint, channelCredentials);
```

I could have just wrote:
```
 // Create the channel using S2A-secured channel credentials.
builder = Grpc.newChannelBuilder(endpoint, channelCredentials);
```